app/questionnaire/routing/operations.py

```diff
@@ -81,18 +98,39 @@ def evaluate_any_in(lhs: Sequence, rhs: Sequence) -> bool:
 
     @staticmethod
     def resolve_date_from_string(
-        date_string: Optional[str], offset: Optional[dict[str, int]] = None
+        date_string: Optional[str],
+        offset: Optional[DateOffset] = None,
+        offset_by_full_weeks: bool = False,
```

Comment:
This flag is needed cause the placeholder date bound works slightly differently, as it always uses a full week offset, so it can do things like "Last week Thursday" if the reference date is less than 1 week ago. This new function should allow us to do everything we need to do.